pkgs/development/libraries/libaom/default.nix

```diff
@@ -35,6 +35,8 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
     # CPU detection isn't supported on Darwin and breaks the aarch64-darwin build:
     "-DCONFIG_RUNTIME_CPU_DETECT=0"
+  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
+    "-DAS_EXECUTABLE=${stdenv.cc.targetPrefix}as"
```

Comment:
Would it work when we set this unconditional?

Reply:
It doesn't, unfortunately. But we could give a full path here, that should work.

Is that a better approach?

Reply:
not really.

Reply:
I though we could do something like https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/mailreaders/mmh/default.nix#L15